<a href="https://colab.research.google.com/github/RSMPAVAN/Python-for-Data-Science/blob/master/Numeric_Operators.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [38]:
#declere variables.

a = 10
b = 10
c = 0

#max length of an integer is upto 18 digits. 

#lets get the values from the user..  we use input function to get the values from the user with a prompt. value inside the quotes is text to be prompoted to the user.

a = input('Enter value a')
b= input('Enter value b')
#return the type of a & b as str. because input function return any value in str format only.
#any value from input /network will be in string . if we want to do any maths with that value 
# we have to convert that into a required format.
#we have to use type casting functions to change the type & perform the neccesary operations

#int/float to string - str(a)
#string to int / float  - int(a) or float(a) ##""works only when all charecters in the string is numbers. if any alphabets is present - it will throw you an error.""

print(type(a),type(b))

a = int(input('Enter a'))
b = int(input('Enter b'))
#type will int - using a type casting function we changed the type of a,b from str to int 

print(type(a),type(b),a,b)


Enter value a10
Enter value b10
<class 'str'> <class 'str'>
Enter a20
Enter b20
<class 'int'> <class 'int'> 20 20


In [36]:
#Addition  or concatenate  + 
# + symbol works as a addition if both values are in int / float
# + symbol works as concatentate ( put together) if both values are strings
#if one value is int / float & another is string - it will throws Type error - 

x = 'hello'
y= 'world'

z = x+y
c = a + b  
print(c,z)
y= '10'
print(c + int(y))



20 helloworld
30


In [6]:
#multiplications *

c = a*b

print(c)

100


In [7]:
#Substraction - 

c = a-b
print(c)

0


In [12]:
#Remainder %. very helpfull when u need to find whether a number is even  or odd.. 

c = a%b
print(c)

0


In [9]:
# power **

c = a**b
print(c)

10000000000


In [10]:
#division.  / - normal division . return the values along with decimal point also (if any ). also called as floting point division

c = a/b
print(c)



1.0


In [11]:
#division - // - floor division - return the nearst roud off integer value. also called as Interger division.

c = a//b
print(c)

1


Observation : // return the nearst integer value 
               / return the floating point value.

Order of **Evaluation**. As like in Maths, some operators are more powerful than others.  Python follow an order called "operator precedence"

In [27]:
#consider an expression. it will results to 6.99. but how it got calulated. 

x = 1 + 2 * 3 - 4 /  5**6

print(x)


6.999744


Paratheses has highest respect. follows by Power . follow by Multiplication / dvision / remainder  , follow by addition & substraction , follow by left to right.

() - highest proiority 
**  - second highest proiority
*, / , % - third highest proiority
+,-  fourth highest priority.

so based on the proioirty - our expression will evaluated from left to right.

in the above example - ** in the right hand has high priority - 
5 **6 = 15625

then division takes next prioirty - 4 / 15625 = 0.000256

then multiplication takes next - 2 * 3 = 6

then substraction takes next = 6 - 0.000256  = 5.999744

then Addition takes next = 1 +  5.999744  = 6.999744



In [29]:
#consider the expression - it will results u 11.0 - 

x = 1 + 2 ** 3 / 4 * 5
#step 1 :  2** 3 = 8
#step 2 :  8 /4 = 2.0
#step 3 : 2.0 * 5 = 10.0
#step 4: 1 + 10.0 = 11.0

#re write the same expression with parathesis as per python execution - 1 +( ((2 **3)/4)*5)
print(x)

11.0


In [33]:
#type of a variable. 

# we have different types of constants - integer constant, floating point constant & string constant
#it always important to what kind of value we are storing in the varaiable

#we have type(variable_name) to know the type of the variable.

#in python we dont declere type of the variable during the intilization .
 #so we can change the type of the varaiable at any time at any place. 
 #same variable name can be used for differrent type of values. but memory holds the most latest one only.
 #first c is an int, later to float , later to string - if we print c, now value will be 'Hello!'
 

type(c)
#manuplate - c now to float
c = 2.0
type(c)
#manuplate - c now to string 
c = "hello!"
type(c)

str